engines/suppliers/db/migrate/20150825155330_create_order.rb

```diff
@@ -1,13 +1,35 @@
 class CreateOrder < ActiveRecord::Migration
-  def change
-    create_table :orders do |t|
-      t.integer :user_id, null: false
-      t.integer :group_id, null: false
+  def up
+    execute <<-SQL
+      CREATE TABLE orders (
+        id             serial  PRIMARY KEY,
+        from_user_id   integer REFERENCES users,
+        from_group_id  integer REFERENCES groups,
+        to_group_id    integer REFERENCES groups,
+        to_producer_id integer REFERENCES producers,
+        confirm_before timestamp without time zone NOT NULL,
+        created_at     timestamp without time zone NOT NULL,
+        updated_at     timestamp without time zone NOT NULL,
+        CHECK(
+          (
+            (from_user_id IS NOT NULL)::integer +
+            (from_group_id IS NOT NULL)::integer
+          ) = 1
+        ),
+        CHECK(
+          (
+            (to_group_id IS NOT NULL)::integer +
+            (to_producer_id IS NOT NULL)::integer
+          ) = 1
+        )
+      );
```

Comment:
Yes, but we need to understand that our db/schema.rb won't be in sync and that can be troublesome. We better find out where/what relies on it.